### ISSUE TITLE
[js] js.html externs refer deprecated js.Promise type

### ISSUE THREAD

**Issue by nadako** [HaxeFoundation/haxe#8282]
Try compiling for JS with `--macro include("js.html")` and you see some deprecation warnings due to externs using unqualified `Promise` access which resolves to `js.Promise`, which is now deprecated. The externs should import `js.lib.Promise` or specify the full path.

**Comment by Aurel300** [comment]
This was fixed with c694332

**Comment by Gama11** [comment]
I don't think so, I think that was a direct modification of the generated externs as a temporary workaround? The extern generator would still need to be updated.

**Comment by RealyUniqueName** [comment]
Yes. I fixed directly to get rid of warnings and left this issue open because of extern generator is not updated.

**Comment by haxiomic** [comment]
I've removed this from 4.0 because the only issue now is in the generator and it shouldn't block 4.0
